File: wrappers/java/src/VirgilStreamDataSink.java
package com.virgilsecurity.crypto;

public class VirgilStreamDataSink extends VirgilDataSink implements java.io.Closeable {
    private java.io.OutputStream stream;

    public VirgilStreamDataSink(java.io.OutputStream stream) {
        this.stream = stream;
    }

    @Override
    public boolean isGood() throws java.io.IOException {
        // If stream is not good, method 'write' will throw exception.
        return true;
    }

    @Override
    public void write(byte[] data) throws java.io.IOException {
        this.stream.write(data, 0, data.length);
    }

    @Override
    public void close() throws java.io.IOException {
        this.stream.close();
        this.delete();
    }
}

File: wrappers/net/src/VirgilStreamDataSink.cs
using virgil.crypto;

namespace virgil.crypto {

public class VirgilStreamDataSink : VirgilDataSink
{
    private readonly System.IO.Stream stream;

    public VirgilStreamDataSink(System.IO.Stream target)
    {
        this.stream = target;
    }

    public override bool IsGood()
    {
        return this.stream.CanWrite;
    }

    public override void Write(byte[] data)
    {
        this.stream.Write(data, 0, data.Length);
    }
}

}
